### DISCUSSION TITLE
HDF5 openPMD output too slow with MPI

### DISCUSSION THREAD

**Discussion by jiecaix** [BLAST-WarpX/warpx#4764] [Q&A]
When utilizing HDF5 as the backend for OpenPMD in MPI mode, the writing speed of the output file is approximately 0.1 MB/s, which is significantly slower compared to its performance in OpenMP mode. Despite numerous attempts to recompile the HDF5 and OpenMPI libraries, the efficiency of MPI HDF5 file output speed remains subpar. Conversely, when testing the output speed of ADIOS, it performs admirably with almost negligible waiting time.

![image](https://github.com/ECP-WarpX/WarpX/assets/56915687/569fd9b3-8c72-4d72-8f9d-069ff6bcc232)

![image](https://github.com/ECP-WarpX/WarpX/assets/56915687/62c04c91-7633-4cd3-995d-05feb55a8e26)

**Comment by ax3l** [comment]
Hi @zhazhajust,

Can you share more details about your system, run size and input options?

Guessing into the blue, I would say you need probably more system-specific I/O hints. But please share more details so we can take a look.

Alternatively, simply use ADIOS2 (`bp`) as output and you should see an immediate performance gain.

**Comment by jiecaix** [comment]
This is the machine information.
```
CPU: Intel(R) Xeon(R) Platinum 9242 CPU @ 2.30GHz (96 cores)
Memory: 376 G
OS: CentOS Linux release 7.8.2003 (Core)
Kernel: 3.10.0-1127.el7.x86_64
```
The simulation is a simple LWFA simulation of 2432*896 grids with 1 particle in each cell.
The openpmd output setting is just backend choosing.
```
field.diag_type = Full
field.fields_to_plot = Ex Ey Ez rho
field.file_prefix = ./FieldDiagnostic/
field.format = openpmd
field.intervals = 200
field.openpmd_backend = h5
field.write_species = 0
```
And i had test the adios2, it works well, costing about 5% in the profile output file.
